src/Writers/LaTeXWriter.jl

```diff
@@ -273,9 +273,11 @@ function latex(io::IO, vec::Vector, page, doc)
 end
 
 function latex(io::IO, anchor::Anchors.Anchor, page, doc)
-    id = string(hash(string(anchor.id, "-", anchor.nth)))
-    _println(io, "\n\\hypertarget{", id, "}{}\n")
+    # TODO: currently ignoring anchor.nth
+    # id = string(hash(string(anchor.id, "-", anchor.nth)))
+    id = string(hash(anchor.id))
```

Comment:
I believe we need to keep using `.nth` here. The [CI artifacts here](https://github.com/JuliaDocs/Documenter.jl/actions/runs/2895291989) contain the PDFs from the `test/examples` builds that do use at-index and at-contents blocks, and it looks like at-contents links are currently still broken, which I think is due to this.

Reply:
If the anchors have `.nth` (at least if `> 1`), then it must also appear in the links. Is this currently the case? I couldn't see anything like this in `LaTeXWriter.jl`. (My understanding of the whole package is very limited at present.)

Reply:
In short, yes, it should be used when generating the labels. But in the PDF, there is no point in special casing the `.nth == 1` case, since the link labels are not user-readable anyway.

Reply:
I believe that link ids should use `.nth`, but currently it's not the case. (That's why the links didn't work.) How would one add that to, say, https://github.com/JuliaDocs/Documenter.jl/blob/6c92d59a19348b9ec4f1936bdce91d34511e4c18/src/Writers/LaTeXWriter.jl#L686-L690 in the function `latexinline(io::IO, md::Markdown.Link)`?

Reply:
Right, so we don't really have access to the underlying `Anchor` or docstring objects when we're generating the links. I'd say we just make sure that the anchors we generate in `LaTeXWriter` just hash a string that matches the URL fragment, which is generated here:

https://github.com/JuliaDocs/Documenter.jl/blob/6c92d59a19348b9ec4f1936bdce91d34511e4c18/src/CrossReferences.jl#L98
https://github.com/JuliaDocs/Documenter.jl/blob/6c92d59a19348b9ec4f1936bdce91d34511e4c18/src/CrossReferences.jl#L153

Reply:
But the question remains: How do we figure out what `nth` should be? It seems to me that we have to look up the `AnchorMap`s `doc.internal.headers` and `doc.internal.docs` where `doc::Document` is the first argument of `LaTeXWriter.render`. (`doc` is already carried along in quite a few `latex` functions.) Would you agree?

Reply:
I don't believe that's necessary -- the `.nth` is already included in the URL fragment, no?

https://github.com/JuliaDocs/Documenter.jl/blob/6c92d59a19348b9ec4f1936bdce91d34511e4c18/src/CrossReferences.jl#L98
https://github.com/JuliaDocs/Documenter.jl/blob/6c92d59a19348b9ec4f1936bdce91d34511e4c18/src/Anchors.jl#L130-L137

Also, for what it's worth, it looks like for docstrings we ignore `.nth`, since for those we shouldn't get duplicates. And the `latex(io::IO, anchor::Anchors.Anchor, page, doc)` method only generated anchors/targets for headings, not docstrings, so we don't need to worry about distinguishing those two cases here.

Reply:
Maybe I'm missing something. Here is what I understand: We need to generate the `nth` both for the target and for the link. For the target it's easy. For the link we have (at least) the code snipped from `latexinline(io::IO, md::Markdown.Link)` that I copied above. As far as I can tell, that code is used both for referring to headers and to docstrings. In that code the `nth` is not included at present for otherwise links would work in the PDF file. So we have to add it for headers. The code snippets you have given above both use an argument of type `Anchor`, but we currently don't have an `Anchor` when we generate the link.

Reply:
Unless I am misreading the code myself, `latexinline(io::IO, md::Markdown.Link)` hashes the fragment part of `md.url`, which is determined by the `fragment()` function in `CrossReferences`. This is means that it already includes `.nth` (well, if `.nth` is `>= 2`). We just have to make sure that the string we hash when we generate the targets (and I guess also the at-contents block entires) is consistent with what `CrossReferences` puts into the `.url` field of the `Link` objects.

Basically, I don't think we need to make any changes to the `latexinline(io::IO, md::Markdown.Link)` method.

Reply:
Isn't the reason the links are not working right now that when we generate the targets, we use `.nth` in the string we hash even when `.nth == ``. But it that case it is not included in the fragment, which is what we hash in the link generation method?